readthedocs/builds/models.py

```diff
@@ -301,6 +301,40 @@ def get_bitbucket_url(self, docroot, filename, source_suffix='.rst'):
         )
 
 
+class APIVersion(Version):
+
+    """Version proxy model for API data deserialization
+
+    This replaces the pattern where API data was deserialized into a mocked
+    :py:cls:`Version` object. This pattern was confusing, as it was not explicit
+    as to what form of object you were working with -- API backed or database
+    backed.
+
+    This model preserves the Version model methods, allowing for overrides on
+    model field differences. This model pattern will generally only be used on
+    builder instances, where we are interacting solely with API data.
+    """
+
+    project = None
+
+    class Meta:
+        proxy = True
+
+    def __init__(self, *args, **kwargs):
+        self.project = APIProject(**kwargs.pop('project', {}))
+        # This was replicated from readthedocs.projects.utils.make_api_project.
+        # I'm not certain why these need to be deleted
+        for key in ['resource_uri', 'absolute_url', 'downloads']:
```

Comment:
Because they are fake objects that aren't on the model, but created by the API.

Reply:
Could also invent the logic and delete anything that isn't a field on the project.

Reply:
ah gotcha, i recognized these as model attributes without looking. makes sense.

Ultimately, I think we want to some logic that uses the serializer to model these fields modeling for API consumption, and drop Project model proxying altogether. I'm just not sure what the connection to the serializer looks like yet. I think this is probably the best way to reduce duplication of data however.